silk/tools/otns_manager.py

```diff
@@ -0,0 +1,464 @@
+      vis_y (int): visualization y coordinate.
+      dispatcher_host (str): host address of the OTNS dispatcher.
+      dispatcher_port (int): port number of the OTNS dispatcher.
+      grpc_client (GRpcClient): gRPC client instance from the manager.
+      logger (logging.Logger): logger for the node.
+    """
+    assert node_id > 0
+
+    self.node_id = node_id
+
+    self.sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    self.source_addr = (dispatcher_host, dispatcher_port + node_id)
+    self.dest_addr = (dispatcher_host, dispatcher_port)
+    self.sock.bind(self.source_addr)
+    self.logger = logger
+
+    self.vis_x = vis_x
+    self.vis_y = vis_y
+    self.grpc_client = grpc_client
+
+    self.extaddr = 0xffffffffffffffff
+    self.has_created_otns_node = False
+
+  def close(self):
+    """Close the node's socket connection.
+    """
+    self.sock.close()
+
+  def send_event(self, event_packet: bytes):
+    """Send event bytes packet.
+
+    Args:
+      event_packet (bytes): pacakged event content in bytes.
+    """
+    self.sock.sendto(event_packet, self.dest_addr)
+
+  def send_extaddr_event(self):
+    """Send extaddr event.
+    """
+    event = Event.status_event("extaddr={:016x}".format(self.extaddr))
+    self.logger.debug(
+        "Node {:d} sending extaddr={:016x}".format(self.node_id, self.extaddr))
+    self.sock.sendto(event.to_bytes(), self.dest_addr)
+
+  def update_extaddr(self, extaddr: int):
+    """Update the node's extended address.
+
+    Args:
+        extaddr (int): new extended address of the node.
+    """
+    if extaddr != self.extaddr:
+      if not self.has_created_otns_node:
+        self.create_otns_node()
+
+      self.extaddr = extaddr
+      self.send_extaddr_event()
+
+  def create_otns_node(self):
+    """Call gRPC client to create a note on server for itself.
+    """
+    self.logger.debug("Adding node {:d} to OTNS".format(self.node_id))
+    self.grpc_client.add_node(self.vis_x, self.vis_y, self.node_id)
+    self.has_created_otns_node = True
+
+
+class WpantundOtnsMonitor(signal.Subscriber):
+  """OTNS log monitor for wpantund process logs.
+
+  Attributes:
+    node (OtnsNode): OTNS node instance that handles UDP messaging.
+  """
+
+  node = None
+
+  def process_log_line(self, line: str):
+    """Process a single line of wpantund log.
+
+    Args:
+        line (str): line of log to process.
+    """
+    status_match = re.search(RegexType.STATUS.value, line)
+    if status_match:
+      message = status_match.group(2)
+
+      if self.node is not None:
+        event = Event.status_event(message)
+        self.node.send_event(event.to_bytes())
+
+    get_extaddr_info_match = re.search(RegexType.GET_EXTADDR_RES.value, line)
+    if get_extaddr_info_match:
+      extaddr = get_extaddr_info_match.group(1)
+      if self.node is not None:
+        self.node.update_extaddr(int(extaddr, 16))
+
+  def subscribeHandle(self, sender, **kwargs):
```

Comment:
If `line` argument is always required, we can use:
```suggestion
  def subscribeHandle(self, sender, line, **kwargs):
```

Reply:
I like this but LGTM.com complains about it, so I'll revert to the old way...